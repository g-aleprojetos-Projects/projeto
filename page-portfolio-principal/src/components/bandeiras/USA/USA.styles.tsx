import styled from 'styled-components';
import sizes from 'resourses/sizes';
import colors from 'resourses/colors';

export interface PropsUSA {
  idiomaSelecionado: boolean;
}

export const Container = styled.div<PropsUSA>`
  display: flex;
  width: ${sizes.px55};
  height: 100%;
  margin: ${sizes.px1} ${sizes.px3};
  cursor: ${props => (props.idiomaSelecionado ? 'default' : 'pointer')};
`;

export const PathVermelho = styled.path<PropsUSA>`
  fill: ${props =>
    props.idiomaSelecionado ? `${colors.coral}` : `${colors.diamont}`};
`;

export const PathAzul = styled.path<PropsUSA>`
  fill: ${props =>
    props.idiomaSelecionado ? `${colors.pantone}` : `${colors.cinzento}`};
`;

export const PathBranco = styled.path.attrs({
  fill: `${colors.white}`,
})``;
