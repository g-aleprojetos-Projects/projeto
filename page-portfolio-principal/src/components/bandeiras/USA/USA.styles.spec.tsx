import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as S from './USA.styles';

describe('USA.styles', () => {
  test('Container DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false', () => {
    const container = renderer
      .create(<S.Container idiomaSelecionado={false} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('Container DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true', () => {
    const container = renderer
      .create(<S.Container idiomaSelecionado={true} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('PathVermelho DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false', () => {
    const container = renderer
      .create(<S.PathVermelho idiomaSelecionado={false} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('PathVermelho DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true', () => {
    const container = renderer
      .create(<S.PathVermelho idiomaSelecionado={true} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('PathAzul DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false', () => {
    const container = renderer
      .create(<S.PathAzul idiomaSelecionado={false} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('PathAzul DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true', () => {
    const container = renderer
      .create(<S.PathAzul idiomaSelecionado={true} />)
      .toJSON();
    expect(container).toMatchSnapshot();
  });

  test('PathBranco DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.PathBranco />).toJSON();
    expect(container).toMatchSnapshot();
  });
});
