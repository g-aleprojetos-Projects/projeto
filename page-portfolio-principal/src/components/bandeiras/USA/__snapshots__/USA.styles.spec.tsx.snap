// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`USA.styles Container DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 3.438rem;
  height: 100%;
  margin: 0.063rem 0.188rem;
  cursor: pointer;
}

<div
  className="c0"
/>
`;

exports[`USA.styles Container DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 3.438rem;
  height: 100%;
  margin: 0.063rem 0.188rem;
  cursor: default;
}

<div
  className="c0"
/>
`;

exports[`USA.styles PathAzul DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false 1`] = `
.c0 {
  fill: #3C3C3C;
}

<path
  className="c0"
/>
`;

exports[`USA.styles PathAzul DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true 1`] = `
.c0 {
  fill: #41479B;
}

<path
  className="c0"
/>
`;

exports[`USA.styles PathBranco DEVE ser igual ao snapshot 1`] = `
<path
  className=""
  fill="#ffffff"
/>
`;

exports[`USA.styles PathVermelho DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a false 1`] = `
.c0 {
  fill: #9B9C9A;
}

<path
  className="c0"
/>
`;

exports[`USA.styles PathVermelho DEVE ser igual ao snapshot QUANDO "idiomaSelecionado" for igual a true 1`] = `
.c0 {
  fill: #FF4B55;
}

<path
  className="c0"
/>
`;
